homeassistant/components/samsungtv/remote.py

```diff
@@ -49,3 +50,14 @@ async def async_send_command(self, command: Iterable[str], **kwargs: Any) -> Non
 
         for _ in range(num_repeats):
             await self._bridge.async_send_keys(command_list)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the remote on."""
+        if self._turn_on_action:
+            await self._turn_on_action.async_run(self.hass, self._context)
+        elif self._mac:
+            await self.hass.async_add_executor_job(self._wake_on_lan)
+        else:
+            raise HomeAssistantError(
+                f"Entity {self.entity_id} does not support this service."
+            )
```

Comment:
Does it make sense to add this exception to the `media_player` case as well?

Reply:
I don't think so.

In the media_player, this is handled by `MediaPlayerEntityFeature.TURN_ON` supported feature, and so the corresponding error is already raised.

There is no such supported feature on the remote entity, which is why it needed adding manually.